app/src/main/java/com/keylesspalace/tusky/components/compose/dialog/CaptionDialog.kt

```diff
@@ -77,12 +77,11 @@ fun <T> T.makeCaptionDialog(
     input.filters = arrayOf(InputFilter.LengthFilter(MEDIA_DESCRIPTION_CHARACTER_LIMIT))
 
     val okListener = { dialog: DialogInterface, _: Int ->
-        onUpdateDescription(input.text.toString())
-        withLifecycleContext {
-            onUpdateDescription(input.text.toString())
-                .observe { success -> if (!success) showFailedCaptionMessage() }
+        lifecycleScope.launch {
```

Comment:
how does `lifecycleScope` work if we dismiss it right away?

Reply:
if the scope is cancelled, all calls in it are as well

Reply:
That's what I mean, we fire it and then dismiss the dialog right away which will close the scope

Reply:
It is the scope of the activity, it is fine